provider/marathon/marathon_test.go

```diff
@@ -1801,3 +1807,33 @@ func TestParseIndex(t *testing.T) {
 		})
 	}
 }
+
+func TestMarathonGetBasicAuth(t *testing.T) {
+	provider := &Provider{}
+
+	applications := []struct {
+		application marathon.Application
+		expected    []string
+	}{
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{}},
+			expected: []string{},
+		},
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{
+					"traefik.frontend.auth.basic": "user:password",
+				},
+			},
+			expected: []string{"user:password"},
+		},
+	}
+
+	for _, a := range applications {
+		actual := provider.getBasicAuth(a.application)
+		if !reflect.DeepEqual(actual, a.expected) {
+			t.Fatalf("expected %q, got %q", a.expected, actual)
```

Comment:
Then it should also be fixed here https://github.com/containous/traefik/blob/master/provider/marathon/marathon_test.go#L1371